src/components/pages/Roles/Roles.tsx

```diff
@@ -31,10 +30,14 @@ const COLUMN_DATA: Column[] = [
 
 const Roles = () => {
   const { data, isFetching, isError, error, refetch } = useGetRoles();
-  const [editRoleId, setEditRoleId] = useState<string | null>(null);
-  const { openPortal, closePortal, isOpen, Portal } = usePanelPortal(
-    "is-medium",
-    PanelFormNavigationTitleId,
+  const { generatePanel, openPanel } = usePanel<{
+    editRoleId?: string | null;
+  }>((closePanel, data) =>
+    data?.editRoleId ? (
+      <EditRolePanel roleId={data.editRoleId} close={closePanel} />
+    ) : (
+      <AddRolePanel close={closePanel} />
+    ),
   );
```

Comment:
The idea is that this hook could be used to display the delete confirmation as well e.g.
```typescript
  const { generatePanel, openPanel } = usePanel<{
    editRoleId?: string | null;
    deleteRoles?: string[] | null;
  }>((closePanel, data) => {
    if (data?.editRoleId) {
      return <EditRolePanel roleId={data.editRoleId} close={closePanel} />;
    } else if (data?.deleteRoles) {
      return <DeleteRolesPanel deleteRoles={data.deleteRoles} close={closePanel} />
    }
    return <AddRolePanel close={closePanel} />
  });
```